test/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2UnitTest.java

```diff
@@ -50,23 +52,30 @@
     protected TestCrossSiteScriptV2 createScanner() {
         return new TestCrossSiteScriptV2();
     }
-
+    
+    private String[][] createResponseMatrix(String parameterName, List<String> parameterValues){
+        String[][] responseMatrix = new String[parameterValues.size()][];
+        for (int i = 0; i < responseMatrix.length; i++){
+            responseMatrix[i] = new String[]{parameterName,parameterValues.get(i)};
+        }
+        return responseMatrix;
+    }
+    
     @Test
     public void shouldReportXssInParagraph() throws NullPointerException, IOException {
         String test = "/shouldReportXssInParagraph/";
         
         this.nano.addHandler(new NanoServerHandler(test) {
             @Override
             protected Response serve(IHTTPSession session) {
-                String name = session.getParms().get("name");
+                List<String> nameList = session.getParameters().get("name");
```

Comment:
Probably better to introduce a new class method in `NanoServerHandler` instead (e.g. `protected static String getFirstParamValue(IHTTPSession session, String param)`) to simplify the most common test case, which just uses one parameter with same name.